manifests/config.pp

```diff
@@ -120,5 +120,24 @@
         changes => template('foreman_proxy/sudo_augeas.erb'),
       }
     }
+  } else {
+    # The puppet-agent (puppet 4 AIO package) doesn't create a puppet user and group
+    # but the foreman proxy still needs to be able to read the agent's private key
+    if $foreman_proxy::manage_puppet_group {
+      if !defined(Group[$foreman_proxy::puppet_group]) {
+        group { $foreman_proxy::puppet_group:
```

Comment:
Sorry, I did mean a relationship rather than its physical place in the manifest. The docs for the user type suggest it only autorequires the primary group, not supplementary groups (like puppet_group).